core-lib/src/main/java/org/idpass/smartscanner/lib/ocr/OCRAnalyzer.kt

```diff
@@ -0,0 +1,254 @@
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import androidx.camera.core.ImageProxy
+import com.google.gson.Gson
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import org.idpass.smartscanner.api.ScannerConstants
+import org.idpass.smartscanner.lib.R
+import org.idpass.smartscanner.lib.SmartScannerActivity
+import org.idpass.smartscanner.lib.scanner.BaseImageAnalyzer
+import org.idpass.smartscanner.lib.scanner.config.ImageResultType
+import org.idpass.smartscanner.lib.scanner.config.Modes
+import org.idpass.smartscanner.lib.utils.BitmapUtils
+import org.idpass.smartscanner.lib.utils.extension.cacheImagePath
+import org.idpass.smartscanner.lib.utils.extension.cacheImageToLocal
+import org.idpass.smartscanner.lib.utils.extension.cropCenter
+import org.idpass.smartscanner.lib.utils.extension.encodeBase64
+import org.idpass.smartscanner.lib.utils.extension.isImageBlur
+import org.idpass.smartscanner.lib.utils.extension.setBrightness
+import org.idpass.smartscanner.lib.utils.extension.setContrast
+import java.io.File
+
+open class OCRAnalyzer(
+    override val activity: Activity,
+    override val intent: Intent,
+    override val mode: String = Modes.OCR.value,
+    private val imageResultType: String,
+    private val isShowGuide: Boolean? = false,
+    private val regex: String? = "",
+    private val type: String? = "",
+    private val manualCapture: Boolean = false,
+    private val analyzeStart: Long = 0L
+) : BaseImageAnalyzer() {
+
+    private var captured = false
+    private var startAnalyze = false
+
+    @SuppressLint("UnsafeExperimentalUsageError", "UnsafeOptInUsageError")
+    override fun analyze(imageProxy: ImageProxy) {
+        captured = !manualCapture
+        Handler(Looper.getMainLooper()).postDelayed({
+            startAnalyze = true
+        }, analyzeStart)
+        val bitmap = BitmapUtils.getBitmap(imageProxy)
+        bitmap?.let { bf ->
+            val rotation = imageProxy.imageInfo.rotationDegrees
+            bf.apply {
+                // Increase brightness and contrast for clearer image to be processed
+                setContrast(1.1F)
+                setBrightness(3F)
+            }
+
+            val rectGuide = activity.findViewById<ImageView>(R.id.scanner_overlay)
+            val xGuide = activity.findViewById<View>(R.id.x_guide)
+            val yGuide = activity.findViewById<View>(R.id.y_guide)
+            val viewFinder = activity.findViewById<View>(R.id.view_finder)
+            val capture = activity.findViewById<View>(R.id.manual_capture)
+            var inputBitmap = bf
+            var inputRot = rotation
+            var rotatedBF = BitmapUtils.rotateImage(bf, rotation)
+
+            capture.setOnClickListener {
+                captured = true
+            }
+
+            if (isShowGuide != null && isShowGuide) {
```

Comment:
thats a weird onem what value isshowGuide can take? we cannot check that it is true only? ignoring null and false?

Reply:
Fixed